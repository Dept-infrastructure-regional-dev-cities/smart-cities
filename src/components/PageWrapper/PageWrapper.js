import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Header from '../Header/Header';
import Footer from '../Footer/Footer';
import SideBar from '../SideBar/SideBar';
import style from './PageWrapper.scss';
import {
  NO_CATEGORY,
  NO_CITY,
} from '../../constants';

class PageWrapper extends Component {
  constructor(props) {
    super(props);

    this.state = { isOpen: false };

    this.handleWindowClick = this.handleWindowClick.bind(this);
  }

  handleWindowClick(e) {
    // if the user clicks, and the click is outside the nav, close it
    if (!this.sideBarEl.contains(e.target)) {
      this.toggleNav(false);
    }
  }

  toggleNav(isOpen) {
    this.setState({ isOpen });

    // bind these events on the next tick so that they don't capture the current click
    // (because React delegates events)
    window.setTimeout(() => {
      if (isOpen) {
        window.addEventListener('click', this.handleWindowClick, false);
      } else {
        window.removeEventListener('click', this.handleWindowClick, false);
      }
    });
  }

  render() {
    return (
      <div>
        <div ref={(el) => { this.sideBarEl = el; }}>
          <SideBar
            closeNav={() => this.toggleNav(false)}
            isOpen={this.state.isOpen}
            categoryId={this.props.categoryId}
            cityId={this.props.cityId}
          />
        </div>

        <div className={style.wrapper}>
          <Header
            openNav={() => this.toggleNav(true)}
            cityId={this.props.cityId}
            categoryId={this.props.categoryId}
          />

          <main className={style.main}>
            {this.props.children}
          </main>

          <Footer />
        </div>
      </div>
    );
  }
}

PageWrapper.propTypes = {
  categoryId: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.array,
  ]).isRequired,
  cityId: PropTypes.string,
};

PageWrapper.defaultProps = {
  categoryId: NO_CATEGORY,
  cityId: NO_CITY,
};

export default PageWrapper;
