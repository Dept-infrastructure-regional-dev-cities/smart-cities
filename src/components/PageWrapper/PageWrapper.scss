@import '~style/variables';

.wrapper {
  border: 1px solid blue;
  padding-left: $sidebar_width;
}

.main {
  min-height: 100vh; // stop the footer creeping up the page
}
