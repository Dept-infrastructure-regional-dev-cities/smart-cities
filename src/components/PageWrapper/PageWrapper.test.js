import React from 'react';
import { shallow } from 'enzyme';
import PageWrapper from './PageWrapper';

const defaultProps = {
  categoryId: 'jobs',
  cityId: 'perth',
};

it('should match Snapshot', () => {
  const component = shallow(
    <PageWrapper {...defaultProps}>
      <div>render me</div>
    </PageWrapper>
  );

  expect(component.debug()).toMatchSnapshot();
});
