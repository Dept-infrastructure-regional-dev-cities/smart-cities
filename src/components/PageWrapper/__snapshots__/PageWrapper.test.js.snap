// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
"<div className=\\"pageWrapper\\">
  <div>
    <SideBar closeNav={[Function]} isOpen={false} cityId=\\"perth\\" />
  </div>
  <div>
    <Header openNav={[Function]} cityId=\\"perth\\" categoryId=\\"jobs\\" cityName={[undefined]} categoryColorName={[undefined]} />
    <main className=\\"main\\">
      <div>
        render me
      </div>
    </main>
    <FeedbackButton dimmed={[undefined]} />
    <FooterContainer categoryId=\\"jobs\\" />
  </div>
</div>"
`;
